volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -2489,35 +2489,47 @@ def get_pages(self) -> Iterable[interfaces.objects.ObjectInterface]:
         """
         if not self.i_size:
             return
-        elif not (self.i_mapping and self.i_mapping.nrpages > 0):
+
+        if not (
+            self.i_mapping
+            and self.i_mapping.is_readable()
+            and self.i_mapping.nrpages > 0
+        ):
             return
 
         page_cache = linux.PageCache(
             context=self._context,
             kernel_module_name="kernel",
             page_cache=self.i_mapping.dereference(),
         )
+
         yield from page_cache.get_cached_pages()
 
-    def get_contents(self):
+    def get_contents(self) -> Iterable[Tuple[int, bytes]]:
```

Comment:
This is a little questionable as to whether it's a change in API, but I think it'll be ok because it's just a shift from str to bytes, but we're getting on very thin ice here...